### PR TITLE
Delete modal

### PR DESCRIPTION
**Pull request by ArgoNavis1763**
Delete modal fully functional and all tests working. Edit and delete buttons working only for own comments.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,9 +3,14 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "-": "0.0.1",
```
Comment (TomGroombridge) [inline package.json]: not sure what this is but pretty sure you don't need it

File: src/components/DeleteModal.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
+import { Dialog, Transition } from '@headlessui/react';
+import { ExclamationIcon } from '@heroicons/react/outline';
+import React, { Fragment, useRef } from 'react';
+
+const DeleteModal = (props) => {
+  const handleDelete = () => {
+    const newComments = props.comments.map((comment, index) => {
+      if (comment.id !== props.id) {
+        return comment;
+      } else {
```
Comment (TomGroombridge) [inline src/components/DeleteModal.jsx]: you won't need the else statement now, we filter before setting the new state

File: src/components/DeleteModal.jsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import { Dialog, Transition } from '@headlessui/react';
+import { ExclamationIcon } from '@heroicons/react/outline';
+import React, { Fragment, useRef } from 'react';
+
+const DeleteModal = (props) => {
+  const handleDelete = () => {
+    const newComments = props.comments.map((comment, index) => {
```
Comment (TomGroombridge) [inline src/components/DeleteModal.jsx]: we don't need to define index

File: src/components/delete.jsx [hunk 4]
```diff
@@ -1,19 +1,26 @@
...
+        onClick={() => setOpen(true)}
+        id="delete-button"
       >
         <p>Delete</p>
         <img src="/icons/icon-delete.svg" />
       </button>
+      <DeleteModal
+        setOpen={setOpen}
+        open={open}
+        id={props.id}
```
Comment (TomGroombridge) [inline src/components/delete.jsx]: I think id could be more specific... could this be `commentId`

File: package.json [hunk 5]
```diff
@@ -3,9 +3,14 @@
...
+    "@headlessui/react": "^1.6.4",
+    "@heroicons/react": "^1.0.6",
     "@testing-library/jest-dom": "^5.16.4",
     "@testing-library/react": "^13.3.0",
     "@testing-library/user-event": "^13.5.0",
+    "headless": "^1.2.0",
+    "npm": "^8.12.1",
```
Comment (TomGroombridge) [inline package.json]: no sure you need npm?

**Review by TomGroombridge — COMMENTED** [review]
few comments but looks great in general

File: package.json (hunk 1 above)
Comment (charlielizzy) [reply package.json]: removed

File: src/components/DeleteModal.jsx (hunk 2 above)
Comment (charlielizzy) [reply src/components/DeleteModal.jsx]: removed

File: src/components/DeleteModal.jsx (hunk 3 above)
Comment (charlielizzy) [reply src/components/DeleteModal.jsx]: removed

File: src/components/delete.jsx (hunk 4 above)
Comment (charlielizzy) [reply src/components/delete.jsx]: made more specific

File: package.json (hunk 5 above)
Comment (charlielizzy) [reply package.json]: removed

**Comment by charlielizzy** [comment]
actioned all your comments - will push updated changes to delete modal branch

**Pull request merged**